src/engine/execute.js

```diff
@@ -148,19 +140,19 @@ var execute = function (sequencer, thread) {
             execute(sequencer, thread);
             if (thread.status === Thread.STATUS_PROMISE_WAIT) {
                 return;
-            } else {
-                // Execution returned immediately,
-                // and presumably a value was reported, so pop the stack.
-                currentStackFrame.waitingReporter = null;
-                thread.popStack();
             }
+
+            // Execution returned immediately,
+            // and presumably a value was reported, so pop the stack.
+            currentStackFrame.waitingReporter = null;
+            thread.popStack();
         }
         argValues[inputName] = currentStackFrame.reported[inputName];
     }
 
     // Add any mutation to args (e.g., for procedures).
-    var mutation = blockContainer.getMutation(currentBlockId);
-    if (mutation) {
+    var mutation = blockContainer.getMutation(block);
+    if (mutation !== null) {
```

Comment:
Yes, the old way IS better assuming you want to catch any future case where the `getMutation()` might return undefined... The reason for changing it was simple... In all my test it's quicker to compare to !== null than to allow javascript to do it's casting required to change `mutation` into a boolean. All these changes were about optimising things as best I could, but I am aware that you might prefer to err on the side of caution. The gains are all very slight, but they all add up. What we really need to a good refactor of the runtime engine to reduce the number of function calls as they are proving very expensive. This small tweak is peanuts in the grand scale of things, but it is called for 'every block' so I think is worth considering still? However, it's your call, would you prefer that line reverting?